Ramanujan_joel.java

```diff
@@ -0,0 +1,52 @@
+//
+// Source code recreated from a .class file by IntelliJ IDEA
+// (powered by Fernflower decompiler)
+//
+
+public class Ramanujan {
+  public Ramanujan() {
+  }
+
+  private static boolean isWholeNumber(double error, double n) {
+    return error > Math.abs(n - Math.ceil(n)) || error > Math.abs(n - Math.floor(n));
+  }
+
+  public static boolean isRamanujan(long n) {
+    double error = 1.0E-11D;
+    double maxNum = Math.cbrt((double)n);
+    boolean firstSet = false;
+    double a = 1.0D;
+    double b = 1.0D;
+    double c = 1.0D;
+    double d = 1.0D;
+
+    for(int i = 1; (double)i < maxNum; ++i) {
+      double aCubed = (double)(i * i * i);
+      double resultB = Math.cbrt((double)n - aCubed);
+      if (isWholeNumber(error, resultB) && !firstSet) {
```

Comment:
don't think you need this if statement

Reply:
Yeah will do, I need to lookup more how to more appropriately handle this situation.